azure/app-service-docker-app/blueprint.yaml

```diff
@@ -0,0 +1,145 @@
+
+  - name: TenantID
+    type: Input
+    prompt: What is your existing Azure tenant id?
+
+  # ############################################################################
+  # Location and app details
+  # ############################################################################
+  - name: ResourceGroupLocation
+    type: Select
+    prompt: "Select where this resource group will go:"
+    options:
+      - label: East Asia
+        value: eastasia
+      - label: Southeast Asia
+        value: southeastasia
+      - label: Central US
+        value: centralus
+      - label: East US
+        value: eastus
+      - label: East US 2
+        value: eastus2
+      - label: West US
+        value: westus
+      - label: North Central US
+        value: northcentralus
+      - label: South Central US
+        value: southcentralus
+      - label: North Europe
+        value: northeurope
+      - label: West Europe
+        value: westeurope
+      - label: Japan West
+        value: japanwest
+      - label: Japan East
+        value: japaneast
+      - label: Brazil South
+        value: brazilsouth
+      - label: Australia East
+        value: australiaeast
+      - label: Australia Southeast
+        value: australiasoutheast
+      - label: South India
+        value: southindia
+      - label: Central India
+        value: centralindia
+      - label: West India
+        value: westindia
+      - label: Canada Central
+        value: canadacentral
+      - label: Canada East
+        value: canadaeast
+      - label: UK South
+        value: uksouth
+      - label: UK West
+        value: ukwest
+      - label: West Central US
+        value: westcentralus
+      - label: West US 2
+        value: westus2
+      - label: Korea Central
+        value: koreacentral
+      - label: Korea South
+        value: koreasouth
+      - label: France Central
+        value: francecentral
+      - label: France South
+        value: francesouth
+      - label: Australia Central
+        value: australiacentral
+      - label: Australia Central 2
+        value: australiacentral2
+      - label: South Africa North
+        value: southafricanorth
+      - label: South Africa West
+        value: southafricawest
+
+  - name: AppName
+    type: Input
+    prompt: What do you want to name the application?
+    validate: !expr "regex('[a-zA-Z0-9-]*', AppName)"
+
+  - name: UseSampleApplication
+    type: Confirm
+    prompt: Do you want to deploy a sample application with its database?
+    default: true
+    description: Deploy the sample application from https://github.com/xebialabs/e-commerce-monolith with a MySQL server database
+
+  - name: MySqlMasterPassword
+    type: SecretInput
+    prompt: What is the master password to use for MySQL server that will be provisioned?
+    validate: !expr "regex('^(?=.*?[0-9])(?=.*?[a-z])(?=.*?[A-Z])(?=.*?[-._!#^~@$%*]).{8,128}$', MySqlMasterPassword)"
```

Comment:
Why do you ask for the MySQL password, but default on a username, but don't mention what that username is?

Reply:
Good point, I'll parametrize the user name as well